### ISSUE TITLE
500 Internal Server Error

### ISSUE THREAD

**Issue by el-dias** [xupisco/MoneyLog-Cloud#7]
Desde anteontem, o site apresenta o erro descrito no título acompanhado dessa frase "The server has either erred or is incapable of performing the requested operation."

**Comment by deniscostadsc** [comment]
Acredito que tem a ver com o desligamento da APIv1 do Dropbox.

https://blogs.dropbox.com/developers/2017/09/api-v1-shutdown-details/

**Comment by xupisco** [comment]
Sim, é isso mesmo... estou dando uma olhada, mas mudou bastante coisa, desde o oauth até o appengine. Está mais complicado do que imaginava.

De qq maneira, fiquem a vontade para mandar PR.  
Abs

**Comment by aureliojargas** [comment]
Link da outra issue sobre este problema: #6

**Comment by aureliojargas** [comment]
Valeu @xupisco pelo empenho! Torço pra que você consiga :crossed_fingers: 

Mais alguém quer ajudar? Pode meter a mão, sem medo de ser feliz 😄

**Comment by aureliojargas** [comment]
Atualizei o site do MoneyLog Cloud http://ml.aurelio.net pra informar sobre o problema, e tem um link pra esta issue.

**Comment by douglasmsi** [comment]
Estou tentando aprender sobre autenticação com NodeJS (meio confuso ainda) e estava pensando em implementar algo com o Dropbox pra acessar o moneylog-cloud. 
Tenho uma dúvida.. o moneylog-cloud está hospedado no site do @aureliojargas ... se eu apenas me basear nesse código do @xupisco , saberiam dizer se funcionaria?

**Comment by aureliojargas** [comment]
@douglasmsi a URL do meu site é apenas uma casca, o site de verdade do MoneyLog Cloud está hospedado no Google App Engine. Se você em Node conseguir tratar de ligar com o Dropbox do usuário, e conseguir ler os arquivos TXT dele, é pra funcionar.

**Comment by xupisco** [comment]
Salve @aureliojargas quanto tempo fio... acho que consegui uma alternativa (pelo menos na minha máquina funciona... hahahah), sem usar o GAE, somente com JS (pode hospedar em qq lugar).

Quando puder me chama no hangout para falarmos sobre! Abrassssss....

**Comment by xupisco** [comment]
Fiz até um tema "dark" para modernizar ainda mais o MoneyLog... XD

![chrome_2018-11-07_20-57-27](https://user-images.githubusercontent.com/10261/48166513-ef8e7b00-e2cf-11e8-9f93-eae9b259fbc1.png)

**Comment by el-dias** [comment]
Caramba, eu queria poder testar :open_mouth:

**Comment by xupisco** [comment]
> Caramba, eu queria poder testar 😮

@edlucasdias até dá... interessa?  
Ele ainda não está 100%, observações:

 - Vc já precisa ter os arquivos no dropbox, ainda não funciona para um "novo usuário";
 - Por enquanto não é possível escolher o tema, está como "dark" por padrão;
 - Está hospedado em um ambiente meu (no Firebase);
 - Ainda estou mexendo, coisas estranhas podem acontecer! :)

Se tudo bem pra vc os itens acima, acessa [https://xupisco.net/moneylog](https://xupisco.net/moneylog)  
E depois conta pra gente o que achou!  

[]'s

**Comment by el-dias** [comment]
Eu tava usando a versão com Google Drive, então é só copiar os arquivos para a minha pasta do dropbox que acredito que deve funcionar. Vou experimentar já já! \o/

**Comment by el-dias** [comment]
Comecei a usar desde ontem, o editor do arquivo online funciona bem e já testei tanto em computador com o client do DropBox instalado quanto computador sem ele. To gostando de usar.

O tema ficou estranho (ao menos pra mim), mas acho que é o costume mesmo hahaha

**Comment by xupisco** [comment]
@edlucasdias boa... legal, ainda está faltando o tratamento de erros!  
Sobre o tema, é muito pessoal mesmo... hehehe, por isso normalmente temos trocentos! :)

Vamos esperar o @aureliojargas se pronunciar!
[]'s

**Comment by xupisco** [comment]
Já estou até preocupado com o @aureliojargas ... hehehe, só sei que foi pra Alemenha, será que tem internet lá? XD

**Comment by el-dias** [comment]
Olha, estou satisfeito com essa abordagem até agora. Só falta mesmo criar a estrutura para um novo usuário que por mim estaria pronto hahahaha

**Comment by el-dias** [comment]
Alguma novidade desde então?

**Comment by kafran** [comment]
> Já estou até preocupado com o @aureliojargas ... hehehe, só sei que foi pra Alemenha, será que tem internet lá? XD

Ele deixou de ser servidor público? kkkkk

**Comment by el-dias** [comment]
Estou usando todos os dias, e pra mim está bem estável. Pretende ainda fazer alguma atualização?

**Comment by aureliojargas** [comment]
Fala povo! Desculpem-me pela demora, eu me mudei pra Alemanha faz uns 8 meses e desde então todo meu tempo livre é dedicado a aprender o idioma e me virar por aqui.

@xupisco Que show esse tema escuro, hein? Mandou muito bem. :clap::clap::clap:

Quanto ao código, @xupisco não sei se chegou a ver, mas cerca de um ano atrás eu reescrevi boa parte do código do Moneylog, criando o conceito de plugins de storage (https://github.com/aureliojargas/moneylog/tree/master/storage/drivers). A ideia é ter apenas uma única versão do Moneylog e ela suportar vários tipos de storage (local e remoto). Tudo em JavaScript, rodando no cliente, sem precisar depender do GAE. 

Essa versão nova tá rodando no https://moneylog.aurelio.net, com deploy automático a cada commit novo.

Finalizei isso pouco antes de minha mudança e infelizmente não tive tempo de documentar as novidades nem de atualizar o site do Moneylog. Mas o código está 100%, estou usando o Google Drive como storage e funciona bem. 

Vi que você conseguiu implementar o acesso ao Dropbox somente com JavaScript também, que massa! Tem interesse em colocar isso no formato do plugin de storage? Assim mantemos um único código pro MoneyLog em si e só concentramos esforços onde tem a diferença, que é o storage.

Basicamente seria criar um arquivo `/storage/drivers/dropbox.js` e colocar toda tua lógica ali dentro. A API é bem simples, está descrita no topo do https://github.com/aureliojargas/moneylog/blob/master/storage/index.js. Você pode usar o [plugin do Google Drive](https://github.com/aureliojargas/moneylog/blob/master/storage/drivers/googledrive.js) como exemplo. Nesse plugin do Google Drive eu decidi não implementar a edição de arquivos pra simplificar e não precisar de acesso de escrita aos dados do usuário, mas fique à vontade para implementar a edição se quiser, a API suporta.

Eu vou tentar dedicar um tempo pra atualizar o site do MoneyLog e documentar esse novo esquema. Se precisar de ajuda com o código me pinga no Telegram.

Se não puder/quiser fazer, eu entendo, estamos todos trabalhando por diversão aqui :)

**Comment by el-dias** [comment]
Seria interessante a junção desse método de gravação de Dropbox com o que o Aurelio fez com o GDrive. Eu mesmo prefiro usar o Dropbox por comodidade

**Comment by kafran** [comment]
Alguma novidade quanto a isso? Agora que o [Firefox 68 também endureceu as regras para leitura de arquivos locais]( https://github.com/xupisco/MoneyLog-Cloud/issues/8), seria legal voltar a ter o moneylog na cloud.

**Comment by el-dias** [comment]
Eu continuo usando a implementação do @xupisco e estou bem satisfeito.

**Comment by el-dias** [comment]
Fui tentar logar nesses dias e aparece esse endereço: https://xupisco.net/moneylog/%7B%7B%20auth_url%20%7D%7D

E esse erro na tela:
![image](https://user-images.githubusercontent.com/758283/84109943-42736b80-a9fa-11ea-8ef1-8481ed50975b.png)

**Comment by kafran** [comment]
Há um tempo que já não uso o moneylog cloud mais. O Xupisco e o Aurelio não estão mais mantendo o projeto. O melhor que você faz é usar o moneylog local na máquina dentro de uma pasta no Dropbox.
